probe-rs-tools/src/bin/probe-rs/util/cli.rs

```diff
@@ -505,31 +589,51 @@ impl CliRttClient {
 async fn print_monitor_event(
     rtt_client: &mut Option<impl DerefMut<Target = CliRttClient>>,
     event: MonitorEvent,
+    store_log: &mut StoreLog,
 ) {
-    match event {
+    use tokio::io::AsyncWriteExt;
+
+    let to_log = match event {
         MonitorEvent::RttDiscovered { up_channels, .. } => {
             let Some(client) = rtt_client else {
                 return;
             };
 
             client.on_channels_discovered(&up_channels);
+            None
         }
         MonitorEvent::RttOutput { channel, bytes } => {
             let Some(client) = rtt_client else {
                 return;
             };
 
-            let Some(processor) = client.channel_processors.get_mut(channel as usize) else {
+            let channel = channel as usize;
+            let Some(processor) = client.channel_processors.get_mut(channel) else {
                 return;
             };
 
-            processor.process(&bytes);
+            let written = processor.process(&bytes);
+            Some((
+                ChannelIdentifier::Channel(channel),
+                written.expect("FIXME error handling"),
+            ))
         }
         MonitorEvent::SemihostingOutput(SemihostingOutput::StdOut(str)) => {
-            print!("{}", str)
+            print!("{}", str);
+            Some((ChannelIdentifier::SemihostingStdout, str))
```

Comment:
I think it would be better to NOT special-case the semihosting streams. Technically, they can have any name (depending on what the target actually implements), my `SemihostingOutput` abstraction is not general enough.

Reply:
As in, make it a `ChannelIdentifier::Semihosting(handle)` and alter `SemihostingOutput` to be a `(handle, data)` struct rather than a `handle(data)` tuple?

Reply:
If I understand you correctly, then yes. You don't have to rework SemihostingOutput for, you can just map SemihostingStdout to `ChannelIdentifier::Semihosting(String::from("stdout"))` for example, for now.